### ISSUE TITLE
Nested Form State!

### ISSUE THREAD

**Issue by offwork** [ngxs/store#1073]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => https://github.com/ngxs/store/blob/master/CONTRIBUTING.md
[x] Other... Please describe:
</code></pre>

## Environment

<pre><code>
Libs:
- @angular/core version: 7.0.2
- @ngxs/store version: 3.2.0

Browser:
- [x] Chrome (desktop) version 74
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 10.12.18  
- Platform:  Windows
</code></pre>

## Nested Form State
Use the Sub States in nested form objects as the following possible it?

<pre><code>
export interface DaysOfWeekStateModel {
  items?: string[];
  daysForm: any | null;
}

export interface HoursOfDayStateModel {
  items?: string[];
  hoursForm: any | null;
}

@State<DaysOfWeekStateModel>({
  name: 'DayState',
  defaults: {
    items: [],
    daysForm: {
      model: undefined,
      dirty: false,
      status: '',
      errors: {}
    }
  }
})
export class DaysState {}

@State<HoursOfDayStateModel>({
  name: 'DayState',
  defaults: {
    items: [],
    hoursForm: {
      model: undefined,
      dirty: false,
      status: '',
      errors: {}
    }
  }
})
export class HoursState extends DaysState {}
</code></pre>

**Comment by splincode** [comment]
?

```ts
@State({
  name: 'calendarForm',
  defaults: {
    items: []
  },
 children: [DaysState, HoursState]
})
export class CalendarState {}

@State({
  name: 'daysForm',
  defaults: {
    model: undefined,
      dirty: false,
      status: '',
      errors: {}
  }
})
export class DaysState {}

@State({
  name: 'hoursForm',
  defaults: {
    model: undefined,
      dirty: false,
      status: '',
      errors: {}
  }
})
export class HoursState {}
```

**Comment by offwork** [comment]
😀a little fatigue, a little confusion ...
I'm trying to nested form object with the Sub States but, I was not a success. Is it possible to do this in a similar structure to the code above? There's something similar in the [documents](https://ngxs.gitbook.io/ngxs/advanced/sub-states#caveats).

**Comment by splincode** [comment]
Create stackblitz example for reproduce

**Comment by splincode** [comment]
Released in 3.5.0
